yt_dlp/extractor/piramidetv.py

```diff
@@ -0,0 +1,113 @@
+from .common import InfoExtractor, SearchInfoExtractor
+from ..utils import (
+    traverse_obj,
+    unified_timestamp,
+    url_or_none,
+)
+
+
+class PiramideTVIE(InfoExtractor):
+    _VALID_URL = r'https?://piramide\.tv/video/(?P<id>[\w-]+)'
+    _TESTS = [{
+        'url': 'https://piramide.tv/video/wWtBAORdJUTh',
+        'info_dict': {
+            'id': 'wWtBAORdJUTh',
+            'ext': 'mp4',
+            'title': 'md5:79f9c8183ea6a35c836923142cf0abcc',
+            'description': '',
+            'thumbnail': 'https://cdn.jwplayer.com/v2/media/W86PgQDn/thumbnails/B9gpIxkH.jpg',
+            'channel': 'León Picarón',
+            'channel_id': 'leonpicaron',
+            'timestamp': 1696460362,
+            'upload_date': '20231004',
+        },
+    }, {
+        'url': 'https://piramide.tv/video/wcYn6li79NgN',
+        'info_dict': {
+            'id': 'wcYn6li79NgN',
+            'title': 'ACEPTO TENER UN BEBE CON MI NOVIA\u2026? | Parte 1',
+            'description': '',
+            'channel': 'ARTA GAME',
+            'channel_id': 'arta_game',
+        },
+        'playlist_count': 4,
+    }]
+
+    def _real_extract(self, url):
+        def extract_video(video_id, fatal=False):
+            video_data = self._download_json(f'https://hermes.piramide.tv/video/data/{video_id}',
+                                             video_id, fatal=fatal)
+            formats, subtitles = self._extract_m3u8_formats_and_subtitles(
+                f'https://cdn.piramide.tv/video/{video_id}/manifest.m3u8', video_id, fatal=fatal)
+            video_info = {**traverse_obj(video_data, {
+                'id': ('video', 'id', {str}),
+                'title': ('video', 'title', {str}),
+                'description': ('video', 'description', {str}),
+                'thumbnail': ('video', 'media', 'thumbnail', {url_or_none}),
+                'channel': ('video', 'channel', 'name', {str}),
+                'channel_id': ('video', 'channel', 'id', {str}),
+                'timestamp': ('video', 'date', {unified_timestamp}),
+            }),
+                'formats': formats,
+                'subtitles': subtitles,
+                'webpage_url': f'https://piramide.tv/video/{video_id}',
+                'webpage_url_basename': video_id,
+            }
+            next_video_id = traverse_obj(video_data, ('video', 'next_video', 'id', {str}))
+            return video_info, next_video_id
+
+        video_id = self._match_id(url)
+        entries = []
+        while video_id is not None:
+            video, next_video = extract_video(video_id, (not entries))
+            if video.get('formats'):
+                entries.append(video)
+            video_id = next_video if next_video != video_id else None
```

Comment:
I'm not a huge fan of this logic. This doesn't work with lazy extraction or `--no-playlist`.
Since I can't think of a better way to do this, I would just keep it simple for now and only extract a single video.
If someone comes up with a better way to extract a new PR can be opened.